spikeinterface/preprocessing/highpass_spatial_filter.py

```diff
@@ -0,0 +1,290 @@
+    """
+    Perform destriping with high-pass spatial filtering. Uses
+    the kfilt() function of the International Brain Laboratory.
+
+    Median average filtering, by removing the median of signal across
+    channels, assumes noise is constant across all channels. However,
+    noise have exhibit low-frequency changes across nearby channels.
+
+    This function extends the cut-band from 0 to 0.01 Nyquist to
+    remove these low-frequency changes during destriping.
+
+    Parameters
+    ----------
+    recording : BaseRecording
+        The parent recording
+    n_channel_pad : int
+        Number of channels to pad prior to filtering. 
+        Channels are padded with mirroring. 
+        If None, no padding is applied, by default 5
+    n_channel_taper : int
+        Number of channels to perform cosine tapering on
+        prior to filtering. If None and n_channel_pad is set,
+        n_channel_taper will be set to the number of padded channels.
+        Otherwise, the passed value will be used, by default None
+    direction : str
+        The direction in which the spatial filter is applied, by default "y"
+    agc_options: dict, str, or None
+        Options for automatic gain control. By default, gain control
+        is applied prior to filtering to improve filter performance.
+        When the argument is "ibl", the function will use the IBL pipeline defaults (agc on).
+        Setting to None will turn off gain control, by default None.
+        To customize, pass a dictionary with the fields:
+            * agc_options["window_length_s"] - window length in seconds
+            * agc_options["sampling_interval"] - recording sampling interval
+    butter_kwargs: dict
+        Dictionary with fields "N" and "Wn" to be passed to
+        scipy.signal.butter, by default N=3 and Wn=0.01
+    """
+    name = 'highpass_spatial_filter'
+
+    def __init__(self, recording, n_channel_pad=None, n_channel_taper=5, direction="y",
+                 agc_options=None, butter_kwargs=None):
+        BasePreprocessor.__init__(self, recording)
+
+        # Check single group
+        channel_groups = recording.get_channel_groups()
+        assert len(np.unique(channel_groups)) == 1, \
+            ("The recording contains multiple groups! It is recommended to apply spatial filtering "
+             "separately for each group. You can split by group with: "
+             ">>> recording_split = recording.split_by(property='group')")
+        # If location are not sorted, estimate forward and reverse sorting
+        channel_locations = recording.get_channel_locations()
+        dim = ["x", "y", "z"].index(direction)
+        assert dim < channel_locations.ndim, f"Direction {direction} is wrong"
+        locs_mono = channel_locations[:, dim]
+        if np.array_equal(np.sort(locs_mono), locs_mono):
+            order_f = None
+            order_r = None
+        else:
+            order_f = np.argsort(locs_mono)
```

Comment:
Odering only on dimenssion can be not predictible. We should order by Y and then x to be sure to have the same order always.

Reply:
are you sure it's not predictable?